src/openmc_nek_driver.cpp

```diff
@@ -78,45 +80,32 @@ void OpenmcNekDriver::init_mappings()
   const auto& heat = this->get_heat_driver();
   if (heat.active()) {
     // Get centroids from heat driver
-    elem_centroids_ = heat.centroids();
+    auto elem_centroids = heat.centroids();
 
     // Broadcast centroids onto all the neutronics procs
-    this->get_neutronics_driver().broadcast(elem_centroids_);
-
-    // Step 2: Set element->cell and cell->element mappings
-    // Create buffer to store cell instance indices corresponding to each Nek global
-    // element. This is needed because calls to OpenMC API functions can only be
-    // made from processes
-    std::vector<int32_t> elem_to_cell(heat.n_global_elem());
-
-    if (openmc_driver_->active()) {
-      std::unordered_map<CellInstance, index> cell_index;
-
-      for (int32_t i = 0; i < elem_to_cell.size(); ++i) {
-        // Determine cell instance corresponding to global element
-        Position elem_pos = elem_centroids_[i];
-        CellInstance c{elem_pos};
-
-        // If this cell instance hasn't been saved yet, add it to cells_ and
-        // keep track of what index it corresponds to
-        if (cell_index.find(c) == cell_index.end()) {
-          cell_index[c] = openmc_driver_->cells_.size();
-          openmc_driver_->cells_.push_back(c);
-        }
-        auto i_cell = cell_index.at(c);
-        cell_to_elems_[i_cell].push_back(i);
+    this->get_neutronics_driver().broadcast(elem_centroids);
+
+    // Set element->cell and cell->element mappings. Create buffer to store cell
+    // handles corresponding to each heat-fluids global element.
+    elem_to_cell_.resize(heat.n_global_elem());
+
+    auto& neutronics = this->get_neutronics_driver();
+    if (neutronics.active()) {
+      // Get cell handle corresponding to each element centroid
+      elem_to_cell_ = neutronics.find(elem_centroids);
 
-        // Set value for cell instance in array
-        elem_to_cell[i] = i_cell;
+      // Create a vector of elements for each neutronics cell
+      for (int32_t elem = 0; elem < elem_to_cell_.size(); ++elem) {
+        auto cell = elem_to_cell_[elem];
+        cell_to_elems_[cell].push_back(elem);
       }
 
       // Determine number of OpenMC cell instances
-      n_cells_ = gsl::narrow<int32_t>(openmc_driver_->cells_.size());
+      n_cells_ = cell_to_elems_.size();
     }
 
     // Set element -> cell instance mapping on each Nek rank
-    intranode_comm_.Bcast(elem_to_cell.data(), elem_to_cell.size(), MPI_INT32_T);
-    elem_to_cell_ = elem_to_cell;
+    intranode_comm_.Bcast(elem_to_cell_.data(), elem_to_cell_.size(), MPI_UINT64_T);
```

Comment:
I have some concerns about using `MPI_UINT64_T` here.   `elem_to_cell_` is an opaque `CellHandle` type, but we're making a risky assumption about the type here.  

My main issue is that, in the ENRICO namespace, we're using `gsl::index` as `CellHandle`.  However, there's been some back-and-forth in GSL Lite about whether `gsl::index` is defined as `ptrdiff_t` or`size_t`.  Right now in GSL LIte, it's a `size_t`, but it was a `ptrdiff_t` in the recent past.  Furthermore, I think it's still a `ptrdiff_t` in MS GS.  I think it would be really hard to remember to switch our MPI datatype here (or detect the error) if required by a switch in GSL versions/implementations.  

I would propose (probably in a different PR) that we define our own `MPI_GSL_INDEX_T` macro that's detected at configure time, and then use that macro here instead of `MPI_UINT64_T`.  Thoughts?

Reply:
Actually, I think I have a better solution: `Driver::broadcast` will use the correct MPI type based on the argument that is passed, so if we use that, there is no hardcoded assumption about the type. The only problem right now is that we need to broadcast over the intranode comm (as opposed to the communicator attached to a driver) -- if we move that method to the communicator itself though, then it will be possible to use it.

Reply:
Just pushed a commit that adopts this approach. Let me know what you think